WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_comment_reply_checks( $file_content ) {
+		global $sniff_helper;
+		if ( preg_match( '/wp_enqueue_script\(\s?("|\')comment-reply("|\')/i', $file_content ) ) {
+			$sniff_helper['comment_reply']['enqueued'] = true;
+			$sniff_helper['comment_reply']['comment_reply_term'] = true;
+		} elseif ( preg_match( '/comment-reply/', $file_content ) ) {
+			$sniff_helper['comment_reply']['comment_reply_term'] = true;
+		}
+	}
+
+	/**
+	 * This function checks for comment pagination.
+	 * If found set comment_pagination to true.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_comments_pagination_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'paginate_comments_links' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'the_comments_navigation' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'the_comments_pagination' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'next_comments_link' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+		if ( false !== strpos( $file_content, 'previous_comments_link' ) ) {
+			$sniff_helper['comments_pagination'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks the global content_width is set.
+	 * Checks include enqueue of comment-reply script and useage of the term comment-reply term.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_content_width_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, '$content_width' ) ) {
+			$sniff_helper['content_width'] = true;
+		}
+		if ( false !== strpos( $file_content , '$GLOBALS' . "['content_width']" ) ) {
+			$sniff_helper['content_width'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks if add_editor_style() is called.
+	 * Checks include enqueue of comment-reply script and useage of the term comment-reply term.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_editor_style_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'add_editor_style' ) ) {
+			$sniff_helper['add_editor_style'] = true;
+		}
+	}
+
+	/**
+	 * This functions checks if for get_avatar and wp_list_comments.
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_avatar_checks( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'get_avatar' ) ) {
+			$sniff_helper['avatar_check'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_list_comments' ) ) {
```

Comment:
wp_list_comments uses avatars by default unless explicitly removed as part of $args. It was therefore included as part of the gravatar.php in themecheck. If either 'get_avatar' or 'wp_list_comments' exist it is OK. 

However wp_list_comments may exclude gravatars so there is a small window of a false positive.